### PR TITLE
fix(Scripts/Hyjal): Air Burst should not be cast on MT

### PR DESCRIPTION
**Pull request by elthehablo**
Also changed some things to doomfire targetting. However this will still go wrong as the doomfire dummies are scripted as hostile npcs that just apply the debug on attack swing. Even if we remove the follow logic, they still chase players like normal hostile mobs would. So this needs some more work.

Update: they now chase a random target every 5s. Seems like a good intermediate solution to me until we figure out 100% for sure it's just random movement

## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/7030

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

## Known Issues and TODO List:

- [ ] re-write Doomfire logic

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by heyitsbench** [comment]
> [...] they still chase players like normal hostile mobs would [...]

Maybe `SetReactState`?

**Comment by elthehablo** [comment]
> > [...] they still chase players like normal hostile mobs would [...]
> 
> Maybe `SetReactState`?

That would work. But I am wondering if they indeed don't chase players at all, or simply change their target every so often. This would be something that would be hard to prove tho

File: src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp [hunk 1]
```diff
@@ -186,35 +186,20 @@ struct npc_doomfire_targetting : public ScriptedAI
 
     void Reset() override
     {
-        _chaseTarget = nullptr;
         ScheduleTimedEvent(5s, [&]
         {
-            if (_chaseTarget)
+            // chases someone else every 5 seconds
+            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 0.0f, true, false))
```
Comment (PkllonG) [inline src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp:192]: image

Comment (PkllonG) [reply src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp:192]: void DoCastDoomFire(Unit* target)
```
//hack?
doomfireSpirit->setAttackTimer(BASE_ATTACK, 27000);
doomfire->setAttackTimer(BASE_ATTACK, 27000);
```

Comment (elthehablo) [reply src/server/scripts/Kalimdor/CavernsOfTime/BattleForMountHyjal/boss_archimonde.cpp]: I don't want them to attack. The fire is an immolation like aura on the creatures

**Comment by elthehablo** [comment]
this lines up with sniffs. random movement stops every 1600ms. basically the spirit seems to wander to a different direction every 1600ms

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**